### ISSUE TITLE
Input and output braille table should automatically be the system language

### ISSUE THREAD

**Issue by winman3000** [nvaccess/nvda#16390]
### Is your feature request related to a problem? Please describe.

If a Braille display is selected or automatically recognized by NVDA, the default Braille table is an English Braille table, regardless of which language the system is currently set to. The desired Braille table must still be individually adapted to the user's requirements in the settings.

### Describe the solution you'd like

Similar to the other areas of NVDA, the braille table of the respective system language should be set. I assume that computer Braille is the most commonly used default setting for most languages. If the user wants a different braille table, he can still select it in the braille settings.

### Additional context

There are many beginners who are irritated when they set up a Braille display with NVDA for the first time and are shown a for example non-German Braille table. It would be a great help for these users if NVDA could do the work for them by automatically setting up the Braille table to the system language.

**Comment by nvdaes** [comment]
I'm interested in this request, so I may create a PR for this, according to NVDA's language.

**Comment by SaschaCowley** [comment]
@nvdaes while this would be a great feature to have, note that there are a number of questions you will have to answer with your PR, including:
- Which braille grade to use (grade 1 literary braille would be my assumption)
- Some languages have dialects for which there are different braille tables, but for which NVDA doesn't have different languages
- Some languages, such as English, have multiple braille codes

See issue #290 for further discussion.

**Comment by nvdaes** [comment]
Thanks @SaschaCowley .
I'm thinking in create a dictionary with available locales in NVDA, probably without dialects, associating these locales with a namedtuple named something like DefaultTable with two fields: input and output.
Another approach would be to add two more fields to the BrailleTable class, with sets for input and output languages, though I suppose that the dictionary approach is more easily understandable
When the PR has a prototype,  we can ask in the translators mailing list. I think we can use grade 1 or literary braille in most cases.

**Comment by CyrilleB79** [comment]
Currently, the default is "Unified English Braille Code grade 1"; is there any reason to change this for English?

Before asking on the translators mailing list, it's worth defining the selection rule for the default table, e.g.:
* literary / integral
* 6 or 8 dots if there is a choice?
* if no literary / integral: should we use computer or grade 2?
* if there is only grade 2 available (do not know if it's a real case), should we use English integral instead? Maybe it depends on the languages...

*

**Comment by nvdaes** [comment]
Cyrille wrote:

> Currently, the default is "Unified English Braille Code grade 1"; is there any reason to change this for English?

No, and for now my plan is to use this table as a fallback, in case a language doesn't have a better choice.

For other points, I've created a branch, but I don't have tested it and is in a very initial stage. But I'll create a draft PR now. You may want to review it.

**Comment by nvdaes** [comment]
Seems that that the PR template has been removed, or something is happening on my side, so I'll create the PR as soon as possible.

**Comment by CyrilleB79** [comment]
There is no issue with the PR template.

Could you try to got to https://github.com/nvaccess/nvda/compare/master...nvdaes:nvda:brailleLangs and press the button "Create pull request"?

**Comment by nvdaes** [comment]
Thanks @CyrilleB79 . I've created a draft PR linked to this issue. I'll add a changelog entry and also I need to reflect this in config.conf.
